compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -297,11 +297,34 @@ static bool isMatmulLike(linalg::LinalgOp &linalgOp) {
          linalgOp.getNumParallelLoops() >= 1;
 };
 
-/// Check if `op` is a linalg.reduce or a linalg.generic that has at least one
-/// reduction iterator.
-static bool hasReductionIterator(linalg::LinalgOp &op) {
-  return isa<linalg::ReduceOp, linalg::GenericOp>(op) &&
-         llvm::any_of(op.getIteratorTypesArray(), linalg::isReductionIterator);
+/// This utility is used to check if `op` is appropriate for the vector
+/// reduction pipeline.
+static bool isProjectingReduction(linalg::LinalgOp op) {
+  const auto loopRanges = op.getStaticLoopRanges();
+  const auto loopTypes = op.getIteratorTypesArray();
+  const auto indexingMaps = op.getIndexingMapsArray();
```

Comment:
Please spell these out: https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable

Reply:
Also, we probably don't need `const` on most of the local variables in this file, especially Types/Values/etc.